src/PrestaShopBundle/DependencyInjection/Compiler/CommandAndQueryCollectorPass.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Collects all Commands & Queries and puts them into container for later processing.
+ */
+class CommandAndQueryCollectorPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!in_array($container->getParameter('kernel.environment'), array('dev', 'test'))) {
+            return;
+        }
+
+        $commandsAndQueries = $this->findCommandsAndQueries($container);
+        $container->setParameter('prestashop.commands_and_queries', $commandsAndQueries);
+    }
+
+    /**
+     * Gets command for each provided handler
+     *
+     * @param ContainerBuilder $container
+     *
+     * @return string[]
+     */
+    private function findCommandsAndQueries(ContainerBuilder $container)
```

Comment:
In a 2nd PR we could also get the Handlers name and classes, so that people know what handles the Commands and Queries they are throwing